### PR TITLE
Add GPTNeoXRewardModel

### PR DESCRIPTION
**Pull request by andreaskoepf**
- add specific reward model based on GPTNeoX: GPTNeoXRewardModel
- support HF auto loading after import of `import models.reward_model`
- make model_training a python module

Note: Running `pip install -e .` in `model/` (parent directory of model_training) is required after this PR has been merged.

### CODE REVIEW

File: model/model_training/models/reward_model.py [hunk 1]
```diff
@@ -1,68 +1,94 @@
...
-            cache_dir=cache_dir,
-        )
 
-        self.transformer = transformer
-        self.out_proj = nn.Linear(transformer.config.hidden_size, 1)
+class GPTNeoXRewardModel(GPTNeoXPreTrainedModel):
+    config_class = GPTNeoXRewardModelConfig
+
+    def __init__(self, config):
+        if type(config) == GPTNeoXConfig:
```
Comment (dvruette) [inline model/model_training/models/reward_model.py:42]: I personally prefer `isinstance(config, GPTNeoXConfig)`

**Review by dvruette — APPROVED** [review]
Very nice, looks good! One thing worth considering: It would be straight forward to abstract the model to a general `GPTNeoXForSequenceClassification` by adding `num_labels` as a config option. Just an idea, not necessary.

File: model/model_training/models/reward_model.py (hunk 1 above)
Comment (andreaskoepf) [reply model/model_training/models/reward_model.py:42]: The direct test was done intentionally, since `GPTNeoXRewardModelConfig` is also derived from `GPTNeoXConfig` ... e.g. the line was added to support the case when a normal non-rm model name is specified. I will add a comment.
